moped-editor/src/views/projects/signalProjectTable/SignalProjectTable.js

```diff
@@ -0,0 +1,397 @@
+      cellStyle: typographyStyle,
+      render: entry =>
+        entry.construction_start
+          ? new Date(entry.construction_start).toLocaleDateString("en-US", {
+              timeZone: "UTC",
+            })
+          : "",
+    },
+    {
+      title: "Last modified",
+      field: "last_modified",
+      editable: "never",
+      cellStyle: typographyStyle,
+      render: entry =>
+        new Date(entry.updated_at).toLocaleDateString("en-US", {
+          timeZone: "UTC",
+        }),
+    },
+  ];
+
+  const projectActions = {
+    update: (newData, oldData) => {
+      // initialize update object with old data
+      const updatedProjectObject = {
+        ...oldData,
+      };
+      // Array of differences between new and old data
+      let differences = Object.keys(oldData).filter(
+        key => oldData[key] !== newData[key]
+      );
+
+      // Loop through the differences and assign newData values.
+      differences.forEach(diff => {
+        updatedProjectObject[diff] = newData[diff];
+      });
+
+      // Remove extraneous fields given by MaterialTable that
+      // Hasura doesn't need
+      delete updatedProjectObject.tableData;
+      delete updatedProjectObject.__typename;
+
+      if (
+        newData.project_sponsor.entity_id !== oldData.project_sponsor.entity_id
+      ) {
+        updateProjectSponsor({
+          variables: {
+            projectId: updatedProjectObject.project_id,
+            entityId: newData.project_sponsor.entity_id,
+          },
+        });
+      }
+
+      updateSignalProject({
+        variables: updatedProjectObject,
+      });
+    },
+    cellUpdate: (newData, oldData, rowData, columnDef) => {
+      // if column definition has a custom edit component, use that mutation to update
+      if (columnDef.customEdit === "projectSponsor") {
+        updateProjectSponsor({
+          variables: {
+            projectId: rowData.project_id,
+            entityId: newData.entity_id,
+          },
+        });
+      } else {
+        const updatedProjectObject = {
+          ...rowData,
+        };
+        updatedProjectObject[columnDef.field] = newData;
+
+        // Remove extraneous fields given by MaterialTable that
+        // Hasura doesn't need
+        delete updatedProjectObject.tableData;
+        delete updatedProjectObject.__typename;
+
+        updateSignalProject({
+          variables: updatedProjectObject,
+        });
+      }
+    },
+  };
+
+  const cellEditComponents = {
+    projectSponsor: props => (
+      <Autocomplete
+        value={props.value ?? "None"}
+        defaultValue={"None"}
+        options={entityList}
+        getOptionLabel={e => e.entity_name}
+        onChange={(event, value) => props.onChange(value)}
+        getOptionSelected={(option, value) =>
+          option.entity_id === value.entity_id
+        }
+        renderInput={params => (
+          <TextField {...params} variant="standard" label={null} />
+        )}
+      />
+    ),
```

Comment:
in future issues I'm going to add more custom edit components, as other cells get their own dropdowns/fields to edit and update.